File: smallpict-php/src/Exceptions/AuthenticationException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class AuthenticationException extends SmallPictException
{
    /**
     * @param string $message
     * @param string|null $requestId
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Authentication failed',
        ?string $requestId = null,
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, 'UNAUTHORIZED', 401, $requestId, $details, $previous);
    }
}

File: smallpict-java/src/main/java/com/smallpict/errors/AuthenticationException.java
package com.smallpict.errors;

import java.util.Map;

public class AuthenticationException extends SmallPictException {
    public AuthenticationException(String message) {
        this(message, null, null);
    }

    public AuthenticationException(String message, String requestId, Map<String, Object> details) {
        super(message, "UNAUTHORIZED", 401, requestId, details, null);
    }
}
